integrations/schemas/package-configuration-schema.json

```diff
@@ -0,0 +1,158 @@
+  "type": "object",
+  "properties": {
+    "id": {
+      "type": "string"
+    },
+    "license_finding_curations": {
+      "items": {
+        "$ref": "#/definitions/licenseFindingCurations"
+      },
+      "type": "array"
+    },
+    "path_excludes": {
+      "items": {
+        "properties": {
+          "comment": {
+            "type": "string"
+          },
+          "pattern": {
+            "type": "string"
+          },
+          "reason": {
+            "$ref": "#/definitions/pathExcludeReason"
+          }
+        },
+        "required": [
+          "pattern",
+          "reason"
+        ],
+        "type": "object"
+      },
+      "type": "array"
+    },
+    "vcs": {
+      "$ref": "#/definitions/vcsMatcher"
+    },
+    "source_artifact_url": {
+      "type": "string"
+    }
+  },
+  "definitions": {
+    "licenseFindingCurationReason": {
+      "enum": [
+        "CODE",
+        "DATA_OF",
+        "DOCUMENTATION_OF",
+        "INCORRECT",
+        "NOT_DETECTED",
+        "REFERENCE"
+      ]
+    },
+    "licenseFindingCurations": {
+      "properties": {
+        "comment": {
+          "type": "string"
+        },
+        "concluded_license": {
+          "type": "string"
+        },
+        "detected_license": {
+          "type": "string"
+        },
+        "line_count": {
+          "type": "integer"
+        },
+        "path": {
+          "type": "string"
+        },
+        "reason": {
+          "$ref": "#/definitions/licenseFindingCurationReason"
+        },
+        "start_lines": {
+          "type": [
+            "integer",
+            "string"
+          ]
+        }
+      },
+      "required": [
+        "path",
+        "concluded_license",
+        "reason"
+      ],
+      "type": "object"
+    },
+    "pathExcludeReason": {
+      "enum": [
+        "BUILD_TOOL_OF",
+        "DATA_FILE_OF",
+        "DOCUMENTATION_OF",
+        "EXAMPLE_OF",
+        "OPTIONAL_COMPONENT_OF",
+        "OTHER",
+        "PROVIDED_BY",
+        "TEST_OF",
+        "TEST_TOOL_OF"
+      ]
+    },
+    "vcsMatcher": {
+      "anyOf": [
+        {
+          "required": [
+            "type"
+          ]
+        },
+        {
+          "required": [
+            "url"
+          ]
+        },
+        {
+          "required": [
+            "revision"
+          ]
+        },
+        {
+          "required": [
+            "path"
+          ]
+        }
+      ],
+      "properties": {
+        "path": {
+          "type": "string"
+        },
+        "revision": {
+          "type": "string"
+        },
+        "type": {
+          "type": "string"
+        },
+        "url": {
+          "type": "string"
+        }
+      },
+      "type": "object"
+    }
+  },
+  "required": [
+    "id"
+  ],
+  "oneOf": [
+    {
+      "required": [
+        "vcs"
+      ]
+    },
+    {
+      "required": [
+        "source_artifact_url"
+      ]
+    }
+  ]
```

Comment:
Note: this should be correct according to the JSON Schema spec, but vscode-yaml plugin currently [does not correctly handle it](https://github.com/redhat-developer/vscode-yaml/issues/769). The plugin does not produce error if both `vcs` and `source_artifact_url` are provided.